drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/AbstractKieContainerService.java

```diff
@@ -35,25 +32,14 @@
     private BuildInfoService buildInfoService;
 
     protected KieContainer getKieContainer(Path path) {
-        final KieModule kieModule = moduleService.resolveModule(path);
-        return assertKieContainerIsValid(buildInfoService.getBuildInfo(kieModule).getKieContainer());
-    }
-
-    protected KieContainer getKieContainerClone(Path path) {
-        final KieModule kieModule = moduleService.resolveModule(path);
-        final BuildInfo buildInfo = buildInfoService.getBuildInfo(kieModule);
-
-        if (buildInfo instanceof BuildInfoImpl) {
-            // The kie builder needs to be cloned so that the original does not get altered.
-            final Builder clone = ((BuildInfoImpl) buildInfo).getBuilder().clone();
-            clone.build();
-
-            return assertKieContainerIsValid(clone.getKieContainer());
+        KieModule kieModule = moduleService.resolveModule(path);
+        KieContainer kieContainer = buildInfoService.getBuildInfo(kieModule).getKieContainer();
+        if (kieContainer == null) {
+            throw new IllegalArgumentException("Retrieving KieContainer has failed. Fix all compilation errors within the " +
+                                                       "project and build the project again.");
         }
-
-        throw new IllegalStateException("Failed to clone Builder.");
+        return assertKieContainerIsValid(kieContainer);
```

Comment:
@Rikkola `kieContainer` null check is performed 2 times: line 37 and line 41 calling `assertKieContainerIsValid()`. Please remove one of them.